app/src/main/java/org/thunderdog/challegram/ui/camera/CameraControlButton.java

```diff
@@ -201,7 +201,6 @@ public void onFactorChangeFinished (int id, float finalFactor, FactorAnimator ca
     switch (id) {
       case ANIMATOR_CHANGE: {
         icon = toIcon;
-        toIcon = null;
```

Comment:
Thank you for your contribution. The root cause of the problem is that `onFactorChangeFinished` is called whenever `animator.forceFactor` is called (previously it wasn't the case), and here the code doesn't check for the `finalFactor` value, which triggers the issue. Fixed properly in the commit TGX-Android/Telegram-X@73c15eb.